plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/conversion/StmtConversionVisitor.kt

```diff
@@ -94,7 +94,20 @@ object StmtConversionVisitor : FirVisitor<ExpEmbedding, StmtConversionContext<Re
     }
 
     override fun visitWhenExpression(whenExpression: FirWhenExpression, data: StmtConversionContext<ResultTrackingContext>): ExpEmbedding {
-        val subj = whenExpression.subject?.let { data.convertAndStore(it) }
+        val subj: VariableEmbedding? = whenExpression.subject?.let {
+            val subjExp = data.convert(it)
+            when (val firSubjVar = whenExpression.subjectVariable) {
+                null -> data.store(subjExp)
+                else -> {
+                    // TODO: handle FirVariables uniformly with FirProperties.
+                    data.registerLocalPropertyName(firSubjVar.name)
+                    val subjVar = VariableEmbedding(data.resolveLocalPropertyName(firSubjVar.name), subjExp.type)
+                    data.addDeclaration(subjVar.toLocalVarDecl())
```

Comment:
If you want to fix this TODO immediately I think it makes sense to just have a method in `MethodConversionContext` that creates a new variable, that is do these 3 lines and return the variable.